extend API version information

manabuishii opened the pull request:
extend API version information #1907
https://github.com/galaxyproject/galaxy/issues/1907

extend /api/version

output json file contents with version_major

```json
{
  "extra": {
    "user_tag": "_fix_job_runner_issue",
    "commit-id": "abcde123"
  },
  "version_major": "16.10"
}
```

First check environment value GALAXY_VERSION_JSON_FILE.
if it's not set, try "galaxy root direcotory"/version.json.

if json file is not find, extra set empty.

manabuishii committed:
extend API version information

extend API version information #1907
https://github.com/galaxyproject/galaxy/issues/1907

output json file contents with version_major

```
{"extra": {"user_tag": "_fix_job_runner_issue", "commit-id": "abcde123"}, "version_major": "16.10"}
```

First check environment value GALAXY_VERSION_JSON_FILE.
if it's not set, try "galaxy root direcotory"/version.json.

if json file is not find, extra set empty.

martenson commented:
@manabuishii thanks for the PR, please fix the linting errors reported by Travis.

manabuishii committed:
fix lint

manabuishii committed:
fix lint

martenson commented:
@galaxybot test this

jmchilton committed:
More idiomatic Python for extended version information.

- Eliminate the use of a bare except.
- Use a ``with`` to ensure file handling is cleaned up.

jmchilton reviewed (changes requested):
Thanks for the contribution @manabuishii. Can you cherry-pick https://github.com/jmchilton/galaxy/commit/d744f797bf24e7cc57db42342f19c9ee3d2ed494 into your branch - it cleans up the Python some more.

```
git remote add jmchilton git@github.com:jmchilton/galaxy.git
git fetch jmchilton
git chery-pick d744f797bf24e7cc57db42342f19c9ee3d2ed494
```

martenson commented:
@galaxybot test this
